lib/datadog/statsd/forwarder.rb

```diff
@@ -0,0 +1,111 @@
+# frozen_string_literal: true
+
+module Datadog
+  class Statsd
+    class Forwarder
+      attr_reader :telemetry
+      attr_reader :transport_type
+
+      def initialize(
+        host: nil,
+        port: nil,
+        socket_path: nil,
+
+        buffer_max_payload_size: nil,
+        buffer_max_pool_size: nil,
+        buffer_overflowing_stategy: :drop,
```

Comment:
There are places you renamed this to just `overflowing_stategy` and others you didn't. Forgotten or intentional? Same for buffer_max_pool_size and buffer_max_payload_size.

Reply:
The classes forwarding the configurations prefix it with `buffer_*` so that it's clear that it's addressed to the buffer. But for the buffer itself, the prefix is useless. Can put it back if you think it's better :).